### PR TITLE
[DEV-632] Use Poetry, update dependencies & readme

### PR DESCRIPTION
**Pull request by sajal-g**
- Updates the [fork](https://github.com/LinuxForHealth/gateway-hl7-listener) with Poetry for dependency management
- Removes gradle related files
- Dockerfile updates
- Readme updates
- Added docker-compose for testing
- Added CircleCI config for build and publish

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version: 2.1
+
+orbs:
+  python: circleci/python@1.0.0
+  artifactory-orb: jfrog/artifactory-orb@1.0.1
```
Comment (Palisand) [inline .circleci/config.yml:5]: This orb being used?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          command: |
+            poetry run pytest
+      - run:
+          name: Build python wheel
+          command: |
+            poetry build -f wheel
+      - run:
+          name: Build Docker image
+          command: |
+            docker build -t coverahealth.jfrog.io/development-docker/chip-gateway-hl7-listener:1.0.0 .
```
Comment (Palisand) [inline .circleci/config.yml:33]: Will we update this at some point so it's not always `1.0.0`?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+]
+
+[[tool.poetry.source]]
+name = "development-pypi"
+url = "https://coverahealth.jfrog.io/artifactory/api/pypi/development-pypi/simple"
+
+[tool.poetry.dependencies]
+python = "^3.9"
+hl7 = "^0.4.5"
+asyncio-nats-client = "^0.11.5"
```
Comment (Palisand) [inline pyproject.toml:18]: Nitpick: it would warm my cockles if these and dev deps were sorted alphabetically (but keep `python` at the top)

**Review by Palisand — COMMENTED** [review]
Some minor comments.

To complete my review, I'm going to get this working on my machine.

File: .circleci/config.yml (hunk 1 above)
Comment (sajal-g) [reply .circleci/config.yml]: removed

File: pyproject.toml (hunk 3 above)
Comment (sajal-g) [reply pyproject.toml]: Done

File: .circleci/config.yml (hunk 2 above)
Comment (sajal-g) [reply .circleci/config.yml:32]: Yes this is temporary. Once we have a standard CircleCI config template and base build image, this will resolve itself.

Comment (Palisand) [reply .circleci/config.yml:32]: Wunderbar!

**Review by Palisand — COMMENTED** [review]
## `.python-version`

I use pyenv to manage the multiple versions of python I have. `poetry install` fails for me unless I have a `.python-version` file in the root dir. Do you think this is something we should add?

Example `.python-version`:
```txt
3.9.9
```

This would ensure all devs' local builds are using the same python version. IMO it would be better if we didn't have to specify a minor version but [that's not yet supported](https://github.com/pyenv/pyenv/issues/1946). Related: https://stackoverflow.com/questions/54315206/should-we-gitignore-the-python-version-file

## Unit tests

All tests passed for me but I did get these warnings:
```
src/test/py/test_main.py::test_processed_received_hl7_messages
  /Users/peteralisandratos/Projects/chip-gateway-hl7-listener/src/main/py/hl7_listener/main.py:68: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    hl7_writer.writemessage(hl7_message.create_ack())
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.

src/test/py/test_main.py::test_processed_received_hl7_messages
  /Users/peteralisandratos/Projects/chip-gateway-hl7-listener/src/main/py/hl7_listener/main.py:95: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    hl7_writer.close()
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.

src/test/py/test_main.py::test_processed_received_hl7_messages
  /Users/peteralisandratos/Projects/chip-gateway-hl7-listener/src/main/py/hl7_listener/main.py:74: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    hl7_writer.writemessage(hl7_message.create_ack(ack_code="AR"))
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.

src/test/py/test_main.py::test_processed_received_hl7_messages
  /Users/peteralisandratos/Projects/chip-gateway-hl7-listener/src/main/py/hl7_listener/main.py:90: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    hl7_writer.writemessage(hl7_message.create_ack(ack_code="AE"))
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.

src/test/py/test_main.py::test_processed_received_hl7_messages
  /Users/peteralisandratos/Library/Caches/pypoetry/virtualenvs/chip-gateway-hl7-listener-vG3oeFYD-py3.9/lib/python3.9/site-packages/_pytest/stash.py:104: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    del self._storage[key]
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Address now or later?

<hr/>

Up next: NATS

File: Dockerfile [hunk 4]
```diff
@@ -1,8 +1,8 @@
-FROM python:3.8.6
+FROM python:3.9-slim-buster
```
[the remark this answers is not in the document]
Comment (sajal-g) [reply Dockerfile:1]: Updated to 3.9.9

**Review by Palisand — APPROVED** [review]
I was able to successfully go through the steps in the README 🎉

File: README.md [hunk 5]
```diff
@@ -26,54 +38,79 @@ https://hub.docker.com/_/nats/
 You can find the instructions for the NATS Cli here:
 https://github.com/nats-io/natscli
 
+A `docker-compose` file is also available for standing up NATS Jetstream server.
+
 ### Create the HL7 stream and MESSAGES consumer
```
Comment (taylorrietz) [inline README.md:43]: Nit – not calling it `MESSAGES` anymore right?

File: README.md [hunk 6]
```diff
@@ -26,54 +38,79 @@ https://hub.docker.com/_/nats/
...
 
 Create the HL7 stream with wildcard/unspecified subjects:
 
 ```bash
-nats str add HL7 --subjects "HL7.*" --ack --max-msgs=-1 --max-bytes=-1 --max-age=1y --storage file --retention limits --max-msg-size=-1 --discard=old --max-msgs-per-subject=-1 --dupe-window=2m --replicas=1
+nats str add hl7 --subjects "hl7.*" --ack --max-msgs=-1 --max-bytes=-1 --max-age=1y --storage file --retention limits --max-msg-size=-1 --discard=old --max-msgs-per-subject=-1 --dupe-window=2m --replicas=1
 ```
 
 Create the MESSAGES consumer for the HL7 stream and filter to a subject named "HL7.MESSAGES":
```
Comment (taylorrietz) [inline README.md:51]: Nit – not calling it `MESSAGES` anymore right?

File: README.md (hunk 5 above)
Comment (sajal-g) [reply README.md]: Fixed.

File: README.md (hunk 6 above)
Comment (sajal-g) [reply README.md]: Fixed.

**Pull request merged**